### PR TITLE
feat: check JSON schema in CI

### PR DESCRIPTION
**Pull request by SMillerDev**

### CODE REVIEW

File: tests/jsonschema/bottle.schema.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://formulae.brew.sh/api/bottle",
```
Comment (MikeMcQuaid) [inline tests/jsonschema/bottle.schema.json:3]: Can remove this, it's going away in https://github.com/Homebrew/formulae.brew.sh/pull/702

**Review by MikeMcQuaid — COMMENTED** [review]
Nice work on this. I think this probably needs to go into Homebrew/brew first because that's the place most likely to break this schema.

**Comment by SMillerDev** [comment]
> Nice work on this. I think this probably needs to go into Homebrew/brew first because that's the place most likely to break this schema.

I talked to Rylan about this and am very much on board to do the work there, but since that will requiring moving all the JSON generation into brew I would like to get the current version working first since it'll be the fastest way to avoid shipping a broken API to users.

File: tests/jsonschema/bottle.schema.json (hunk 1 above)
Comment (SMillerDev) [reply tests/jsonschema/bottle.schema.json:3]: Always great to know you put in work for nothing

**Comment by MikeMcQuaid** [comment]
> I talked to Rylan about this and am very much on board to do the work there, but since that will requiring moving all the JSON generation into brew I would like to get the current version working first since it'll be the fastest way to avoid shipping a broken API to users.

Fine with me 👍🏻.

**Comment by apainintheneck** [comment]
This looks really nice. I was thinking we needed some sort of check for the JSON being produced for the API and this is much nicer than adding more tests for `#to_h` to the main Brew repo. Maybe after this gets merged in I can try working on one for casks too.

**Comment by MikeMcQuaid** [comment]
> I was thinking we needed some sort of check for the JSON being produced for the API and this is much nicer than adding more tests for `#to_h` to the main Brew repo.

We will need something like that too because ultimately that's what's going to break this generation given how minimal the logic in this repository is.

File: .github/workflows/scheduled.yml [hunk 2]
```diff
@@ -163,6 +163,14 @@ jobs:
       - name: Validate build
         run: ./script/validate-build.rb
 
+      - name: Validate schemas
+        run: |
+          apt-get install -y jsonschema
```
Comment (SMillerDev) [inline .github/workflows/scheduled.yml:168]:
```suggestion
          sudo apt-get install -y jsonschema
```

File: .github/workflows/scheduled.yml [hunk 3]
```diff
@@ -163,6 +163,14 @@ jobs:
       - name: Validate build
         run: ./script/validate-build.rb
 
+      - name: Validate schemas
+        run: |
+          sudo apt-get install -y jsonschema
```
Comment (SMillerDev) [inline .github/workflows/scheduled.yml:168]:
```suggestion
          sudo apt-get install -y python-jsonschema
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -32,3 +32,11 @@ jobs:
 
       - name: Generate site
         run: bundle exec rake formulae cask
+
+      - name: Validate schemas
+        run: |
+          apt-get install -y jsonschema
```
Comment (SMillerDev) [inline .github/workflows/tests.yml:38]:
```suggestion
          sudo apt-get install -y python-jsonschema
```

**Comment by MikeMcQuaid** [comment]
@SMillerDev Sorry, history-rewriting broke this. Reopened on https://github.com/Homebrew/formulae.brew.sh/pull/751.

**Pull request closed without merging**